File: Year 2/Semester 4/Programming and Projection Environments/Week 8/SeventhWeekCSharpServerClient/networking/utils/AbsConcurrentServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
namespace SeventhWeekCSharpServerClient.networking.utils
{
    public abstract class AbsConcurrentServer: AbstractServer
    {
        public AbsConcurrentServer(string host, int port) : base(host, port)
        { }

        public override void ProcessRequest(TcpClient client)
        {
            Thread t = CreateWorker(client);
            t.Start();
        }

        protected abstract Thread CreateWorker(TcpClient client);
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 10/ProtobuffProjectJavaORM/Networking/src/main/java/gRPCProject/network/utils/AbsConcurrentServer.java
package gRPCProject.network.utils;

import java.net.Socket;


public abstract class AbsConcurrentServer extends AbstractServer {

    public AbsConcurrentServer(int port) {
        super(port);
    }

    protected void processRequest(Socket client) {
        Thread tw=createWorker(client);
        tw.start();
    }

    protected abstract Thread createWorker(Socket client) ;


}
